<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from six_dof_arm.xacro              | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="six_dof_arm" xmlns:xacro="http://ros.org/wiki/xacro">
  <!-- Include materials -->
  <material name="Black">
    <color rgba="0.0 0.0 0.0 1.0"/>
  </material>
  <material name="Red">
    <color rgba="0.8 0.0 0.0 1.0"/>
  </material>
  <material name="White">
    <color rgba="1.0 1.0 1.0 1.0"/>
  </material>
  <!-- BOTTOM FIXED LINK 
This link is the base of the arm in which arm is placed 

-->
  <!-- ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////// -->
  <joint name="bottom_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <parent link="bottom_link"/>
    <child link="base_link"/>
  </joint>
  <link name="bottom_link">
    <visual>
      <origin rpy="0 0 0" xyz=" 0 0 0"/>
      <geometry>
        <box size="1 1 0.02"/>
      </geometry>
      <material name="White"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz=" 0 0 0"/>
      <geometry>
        <box size="1 1 0.02"/>
      </geometry>
    </collision>
    &gt;

  
  </link>
  <gazebo reference="bottom_link">
    <material>Gazebo/White</material>
  </gazebo>
  <!-- ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////// -->
  <!-- BASE LINK -->
  <link name="base_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.024"/>
      <!-- rotate PI/2 -->
      <geometry>
        <box size="0.09 0.09 0.028"/>
      </geometry>
      <material name="Black"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.024"/>
      <!-- rotate PI/2 -->
      <geometry>
        <box size="0.09 0.09 0.028"/>
      </geometry>
    </collision>
    &gt;
	
    <inertial>
      <mass value="1"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="0.5" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>
  <gazebo reference="base_link">
    <material>Gazebo/Black</material>
  </gazebo>
  <joint name="shoulder_pan_joint" type="revolute">
    <parent link="base_link"/>
    <child link="shoulder_pan_link"/>
    <origin rpy="0 0 0" xyz="0 0 0.038"/>
    <axis xyz="0 0 1"/>
    <limit effort="300" lower="-0.95993167404" upper="2.1816457596" velocity="1"/>
    <dynamics damping="50" friction="1"/>
  </joint>
  <!-- ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////// -->
  <!-- SHOULDER PAN LINK -->
  <link name="shoulder_pan_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.0325"/>
      <geometry>
        <box size="0.06 0.02 0.065"/>
      </geometry>
      <material name="Red"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.0325"/>
      <geometry>
        <box size="0.06 0.02 0.065"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="0.5" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>
  <gazebo reference="shoulder_pan_link">
    <material>Gazebo/Red</material>
  </gazebo>
  <joint name="shoulder_pitch_joint" type="revolute">
    <parent link="shoulder_pan_link"/>
    <child link="shoulder_pitch_link"/>
    <origin rpy="0 0 0" xyz="0 0 0.065"/>
    <axis xyz="0 1 0"/>
    <limit effort="300" lower="-1.396122596" upper="1.7453417404" velocity="1"/>
    <dynamics damping="50" friction="1"/>
  </joint>
  <!-- ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////// -->
  <!-- SHOULDER PITCH LINK -->
  <link name="shoulder_pitch_link">
    <visual>
      <origin rpy="1.570795 1.570795 1.570795" xyz="0.0 0 0.0525"/>
      <geometry>
        <box size="0.105 0.057 0.025"/>
      </geometry>
      <material name="White"/>
    </visual>
    <collision>
      <origin rpy="1.570795 1.570795 1.570795" xyz="0.0 0 0.0525"/>
      <geometry>
        <box size="0.105 0.057 0.025"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="0.5" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>
  <gazebo reference="shoulder_pitch_link">
    <material>Gazebo/White</material>
  </gazebo>
  <joint name="elbow_pitch_joint" type="revolute">
    <parent link="shoulder_pitch_link"/>
    <child link="elbow_pitch_link"/>
    <origin rpy="0 0 0" xyz="0 0 0.105"/>
    <axis xyz="0 -1 0"/>
    <limit effort="300" lower="-3.14159" upper="0" velocity="1"/>
    <dynamics damping="50" friction="1"/>
  </joint>
  <!-- ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////// -->
  <!-- ELBOW PITCH LINK -->
  <link name="elbow_pitch_link">
    <visual>
      <origin rpy="1.570795 1.570795 0" xyz="0 0 0.075"/>
      <geometry>
        <box size="0.15 0.025 0.057"/>
      </geometry>
      <material name="Red"/>
    </visual>
    <collision>
      <origin rpy="1.570795 1.570795 0" xyz="0 0 0.075"/>
      <geometry>
        <box size="0.15 0.025 0.057"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="0.5" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>
  <gazebo reference="elbow_pitch_link">
    <material>Gazebo/Red</material>
  </gazebo>
  <joint name="wrist_pitch_joint" type="revolute">
    <parent link="elbow_pitch_link"/>
    <child link="wrist_pitch_link"/>
    <origin rpy="0 0 0" xyz="0.0 0.00 0.150"/>
    <axis xyz="0 1 0"/>
    <limit effort="300" lower="-1.39626222222" upper="1.74532777778" velocity="1"/>
    <dynamics damping="50" friction="1"/>
  </joint>
  <!-- WRIST PITCH LINK -->
  <link name="wrist_pitch_link">
    <visual>
      <origin rpy="0 1.570795 1.570795" xyz="0 0 0.035"/>
      <geometry>
        <box size="0.07 0.025 0.057"/>
      </geometry>
      <material name="Black"/>
    </visual>
    <collision>
      <origin rpy="0 1.570795 1.570795" xyz="0 0 0.035"/>
      <geometry>
        <box size="0.07 0.025 0.057"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="0.5" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>
  <gazebo reference="wrist_pitch_link">
    <material>Gazebo/White</material>
  </gazebo>
  <joint name="gripper_roll_joint" type="revolute">
    <parent link="wrist_pitch_link"/>
    <child link="gripper_roll_link"/>
    <origin rpy="0 0 0" xyz="0 0 0.07"/>
    <axis xyz="0 0 1"/>
    <!--
		<limit effort="300" velocity="1" lower="-2.61799387799" upper="2.6128806087" />
		-->
    <limit effort="300" lower="0" upper="3.14159" velocity="1"/>
    <dynamics damping="50" friction="1"/>
  </joint>
  <!-- ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////// -->
  <!-- GRIPPER ROLL LINK -->
  <link name="gripper_roll_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.005"/>
      <geometry>
        <cylinder length="0.01" radius="0.02"/>
      </geometry>
      <material name="Red"/>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.005"/>
      <geometry>
        <cylinder length="0.01" radius="0.02"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="1"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="0.5" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>
  <gazebo reference="gripper_roll_link">
    <material>Gazebo/Red</material>
  </gazebo>
  <!--joint name="finger_joint1" type="prismatic">
    <parent link="gripper_roll_link"/>
    <child link="gripper_finger_link1"/>
    <origin xyz="0 0 0.01" />
    <axis xyz="1 0 0" />
      <limit effort="100" lower="0" upper="0.025" velocity="1.0"/>


      <safety_controller k_position="20"
                         k_velocity="20"
                         soft_lower_limit="${-0.15 }"
                         soft_upper_limit="${ 0.0 }"/>


    <dynamics damping="50" friction="1"/>
  </joint -->
  <!-- ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////// -->
  <!-- LEFT GRIPPER AFT LINK -->
  <link name="gripper_finger_link1">
    <visual>
      <origin rpy="0 1.570795 0" xyz="0.0 0 0.0535"/>
      <geometry>
        <box size="0.105 0.01 0.01"/>
      </geometry>
      <material name="Black"/>
    </visual>
    <inertial>
      <mass value="1"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="0.5" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>
  <gazebo reference="l_gripper_aft_link">
    <material>Gazebo/Black</material>
  </gazebo>
  <!-- Joint between Wrist roll and finger 1 -->
  <joint name="finger_joint1" type="prismatic">
    <parent link="gripper_roll_link"/>
    <child link="gripper_finger_link1"/>
    <origin xyz="0.0 0.0 0.005"/>
    <axis xyz="1 0 0"/>
    <limit effort="1" lower="0" upper="0.025" velocity="1.0"/>
    <!--    <mimic joint="gripper_finger_joint" multiplier="-1.0" offset="0.0" /> -->
    <dynamics damping="50" friction="1"/>
  </joint>
  <!-- RIGHT GRIPPER AFT LINK -->
  <link name="gripper_finger_link2">
    <visual>
      <origin rpy="0 1.570795 0" xyz="0.0 0 0.0535"/>
      <geometry>
        <box size="0.105 0.01 0.01"/>
      </geometry>
      <material name="Black"/>
    </visual>
    <inertial>
      <mass value="1"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyy="0.5" iyz="0.0" izz="1.0"/>
    </inertial>
  </link>
  <gazebo reference="r_gripper_aft_link">
    <material>Gazebo/Black</material>
  </gazebo>
  <!-- Joint between Wrist roll and finger 2 -->
  <joint name="finger_joint2" type="prismatic">
    <parent link="gripper_roll_link"/>
    <child link="gripper_finger_link2"/>
    <origin xyz="0.0 0.005 0.005"/>
    <axis xyz="1 0 0"/>
    <limit effort="1" lower="-0.025" upper="0" velocity="1.0"/>
    <!--    <mimic joint="gripper_finger_joint" multiplier="-1.0" offset="0.0" /> -->
    <dynamics damping="50" friction="1"/>
  </joint>
  <!-- Grasping frame -->
  <!-- link name="grasping_frame">

      <inertial>
        <origin xyz="0 0 0" rpy="0 0 0"/>
        <mass value="0.0001"/>
        <cuboid_inertia mass="0.0001" x="0.001" y="0.001" z="0.001"/>
        <inertia  ixx="1.0" ixy="0.0"  ixz="0.0"  iyy="100.0"  iyz="0.0"  izz="1.0" />

      </inertial>

    <visual>
      <origin xyz="0 0 0"/>
      <geometry>
					
          <box size="${grasp_frame_len} ${grasp_frame_width} ${grasp_frame_height}" />
					
      </geometry>
      <material name="White" />
    </visual>

    </link>
	
    <joint name="grasping_frame_joint" type="fixed">
      <parent link="gripper_roll_link"/>
      <child link="grasping_frame"/>
      <origin xyz="0.08 0 0" rpy="0 0 0"/>
    </joint>
		
  <xacro:include filename="$(find mastering_ros_robot_description_pkg)/urdf/sensors/xtion_pro_live.urdf.xacro"/ -->
  <!-- ///////////////////////////////////////////////////////////////////////////////////////////////////////////////////// -->
  <!-- Transmissions for ROS Control -->
  <transmission name="tran1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="shoulder_pan_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="tran1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="shoulder_pitch_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="tran1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="elbow_pitch_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="tran1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="wrist_pitch_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="tran1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="gripper_roll_joint">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="tran1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="finger_joint1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="tran1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="finger_joint2">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <!-- ros_control plugin -->
  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="gazebo_ros_control">
      <robotNamespace>/six_dof_arm</robotNamespace>
    </plugin>
  </gazebo>
</robot>

